.github/workflows/node.yml

```
@@ -38,37 +55,65 @@ env:
     CARGO_TERM_COLOR: always
 
 jobs:
-    load-engine-matrix:
-      runs-on: ubuntu-latest
-      outputs:
-        matrix: ${{ steps.load-engine-matrix.outputs.matrix }}
-      steps:
-          - name: Checkout
-            uses: actions/checkout@v4
-
-          - name: Load the engine matrix
-            id: load-engine-matrix
-            shell: bash
-            run: echo "matrix=$(jq -c . < .github/json_matrices/engine-matrix.json)" >> $GITHUB_OUTPUT
+    get-matrices:
+        runs-on: ubuntu-latest
+        outputs:
+            should-run: ${{ steps.get-matrices.outputs.should-run }}
+            engine-matrix-output: ${{ steps.get-matrices.outputs.engine-matrix-output }}
+            host-matrix-output: ${{ steps.get-matrices.outputs.host-matrix-output }}
+        steps:
+            - uses: actions/checkout@v4
+            - id: get-matrices
+              uses: ./.github/workflows/create-tests-matrices
+              with:
+                  language-name: node
+            - name: Echo outputs
+              run: |
+                  echo "should_run: ${{ steps.get-matrices.outputs.should_run }}"
+                  echo "engine-matrix-output: ${{ steps.get-matrices.outputs.engine-matrix-output }}"
+                  echo "host-matrix-output: ${{ steps.get-matrices.outputs.host-matrix-output }}"
+    create-version-matrix:
+        runs-on: ubuntu-latest
+        needs: get-matrices
+        if: ${{ needs.get-matrices.outputs.should-run == 'true' }}
+        outputs:
+            node-matrix-output: ${{ steps.create-version-matrix.outputs.node-matrix-output }}
+        steps:
+            - name: Checkout
+              uses: actions/checkout@v4
+
+            - name: Create version matrix
+              id: create-version-matrix
+              shell: bash
+              # We want to run the full matrix only for scheduled runs and Core changes
+              # For other events we want to run only the always matrix which is first and last version supported
+              run: |
+                  if [[ "${{ github.event_name }}" == "pull_request" || "${{ github.event_name }}" == "push" ]]; then
+                    echo 'node-matrix-output=["16.x","20.x"]' >> $GITHUB_OUTPUT
+                  else
+                    echo 'node-matrix-output=["16.x","17.x","18.x","19.x","20.x"]' >> $GITHUB_OUTPUT
+                  fi
 
     test-ubuntu-latest:
```

```suggestion
    test-node-ubuntu:
```
Following your approach, we need lang name in the job name. If not, please update it for python and java